server/filestore.go

```diff
@@ -5499,19 +5543,26 @@ func (mb *msgBlock) clearCache() {
 		}
 	}
 
-	if mb.cache == nil {
+	mbcache := mb.cache
+	if mbcache == nil {
+		mbcache = mb.ecache.Value()
+	}
+	if mbcache == nil {
 		return
 	}
 
-	buf := mb.cache.buf
-	if mb.cache.off == 0 {
+	buf := mbcache.buf
+	if mbcache.off == 0 {
 		mb.cache = nil
+		mb.ecache.Set(nil)
 	} else {
 		// Clear msgs and index.
-		mb.cache.buf = nil
-		mb.cache.idx = nil
-		mb.cache.wp = 0
+		mbcache.buf = nil
+		mbcache.idx = nil
+		mbcache.wp = 0
 	}
+
+	mb.ecache.Weaken()
```

Comment:
This a no-op if line 5557 excuted?

Reply:
Yes, it would have no effect in that case.